src/Microsoft.FeatureManagement.Telemetry.ApplicationInsights/FeatureManagementBuilderExtensions.cs

```diff
@@ -29,6 +31,8 @@ public static IFeatureManagementBuilder AddApplicationInsightsTelemetryPublisher
                 throw new ArgumentException($"The provided builder's services must not be null.", nameof(builder));
             }
 
+            builder.Services.AddSingleton<ITelemetryInitializer, TargetingTelemetryInitializer>();
```

Comment:
If we add telemetry initialier here. The method name should change to `AddApplicationInsightsTelemetry` or something similar. Adding the telemetry initializer will make all telemetry be tagged with targeting id. Is there any valid reason that the user won't want to have targeting id in some of telemetry?

Reply:
I like @zhiyuanliang-ms suggestion here. `AddApplicationInsightsTelemetry`